core/domain/exp_jobs_one_off_test.py

```diff
@@ -1041,13 +1041,14 @@ def test_for_textangular_validation_and_migration(self):
                 job_id))
 
         # Test that validation fails before migration.
-        self.assertGreater(len(actual_output), 0)
+        self.assertEqual(len(actual_output), 16)
 
         exploration_dict = exploration.to_dict()
         updated_dict = exp_domain.Exploration._convert_v26_dict_to_v27_dict( # pylint: disable=protected-access
             exploration_dict)
         # This is done to ensure that exploration is not passed through CKEditor
         # Migration pipeline.
+        updated_dict['id'] = self.NEW_EXP_ID
```

Comment:
Hi @bansalnitish, can you tell me exactly what is going on when you update this exploration? So that I can add relevant comments.

Reply:
Hi @prasanna08, sorry for the delay in reply. When the exploration is updated, the html strings are converted to a format which is valid with respect to the TextAngular RTE. Also since we use the convert dict function, the exploration schema version is set to 27 and state schema version to 22. Since we don't want this to be further updated to CKEditor format ( as the test is for TextAngular migration ), the version is manually set as 29 and 24 (versions after CKEditor Migration). This is all what happens when we are updating the exploration.